primitives/bridge/src/lib.rs

```diff
@@ -16,3 +138,240 @@ impl Convert<H256, AccountId> for AccountIdConverter {
+	}
+
+	/// API for querying information about the finalized Darwinia headers.
+	///
+	/// This API is implemented by runtimes that are bridging with the Darwinia chain, not the
+	/// Darwinia runtime itself.
+	pub trait DarwiniaFinalityApi {
+		/// Returns number and hash of the best finalized header known to the bridge module.
+		fn best_finalized() -> (BlockNumber, Hash);
+		/// Returns true if the header is known to the runtime.
+		fn is_known_header(hash: Hash) -> bool;
+	}
+	/// Outbound message lane API for messages that are sent to Darwinia chain.
+	///
+	/// This API is implemented by runtimes that are sending messages to Darwinia chain, not the
+	/// Darwinia runtime itself.
+	pub trait ToDarwiniaOutboundLaneApi<OutboundMessageFee: Parameter, OutboundPayload: Parameter> {
+		/// Estimate message delivery and dispatch fee that needs to be paid by the sender on
+		/// this chain.
+		///
+		/// Returns `None` if message is too expensive to be sent to Darwinia from this chain.
+		///
+		/// Please keep in mind that this method returns lowest message fee required for message
+		/// to be accepted to the lane. It may be good idea to pay a bit over this price to account
+		/// future exchange rate changes and guarantee that relayer would deliver your message
+		/// to the target chain.
+		fn estimate_message_delivery_and_dispatch_fee(
+			lane_id: LaneId,
+			payload: OutboundPayload,
+		) -> Option<OutboundMessageFee>;
+		/// Returns dispatch weight, encoded payload size and delivery+dispatch fee of all
+		/// messages in given inclusive range.
+		///
+		/// If some (or all) messages are missing from the storage, they'll also will
+		/// be missing from the resulting vector. The vector is ordered by the nonce.
+		fn message_details(
+			lane: LaneId,
+			begin: MessageNonce,
+			end: MessageNonce,
+		) -> Vec<MessageDetails<OutboundMessageFee>>;
+		/// Returns nonce of the latest message, received by bridged chain.
+		fn latest_received_nonce(lane: LaneId) -> MessageNonce;
+		/// Returns nonce of the latest message, generated by given lane.
+		fn latest_generated_nonce(lane: LaneId) -> MessageNonce;
+	}
+	/// Inbound message lane API for messages sent by Darwinia chain.
+	///
+	/// This API is implemented by runtimes that are receiving messages from Darwinia chain, not the
+	/// Darwinia runtime itself.
+	pub trait FromDarwiniaInboundLaneApi {
+		/// Returns nonce of the latest message, received by given lane.
+		fn latest_received_nonce(lane: LaneId) -> MessageNonce;
+		/// Nonce of latest message that has been confirmed to the bridged chain.
+		fn latest_confirmed_nonce(lane: LaneId) -> MessageNonce;
+		/// State of the unrewarded relayers set at given lane.
+		fn unrewarded_relayers_state(lane: LaneId) -> UnrewardedRelayersState;
+	}
+}
+
+// todo: Reserved for other chains, don't forget change bridge_id
+pub fn derive_account_from_crab_id(id: SourceAccount<AccountId>) -> AccountId {
```

Comment:
These two `derive_*` function seems not used anymore? @xiaoch05 @fewensa

Reply:
Yes, our bridger not have `device_account` command now.
https://github.com/paritytech/parity-bridges-common/blob/d1541e797e6bf16234861eb03b510d5516e0bb44/relays/bin-substrate/src/cli/derive_account.rs#L58